titiler/middleware.py

```diff
@@ -43,3 +48,70 @@ async def dispatch(self, request: Request, call_next):
             f"{timings}, {app_time}" if timings else app_time
         )
         return response
+
+
+class BrotliMode:
+    """Brotli available modes."""
+
+    generic = MODE_GENERIC
+    text = MODE_TEXT
+    font = MODE_FONT
+
+
+class CompressMiddleware:
+    """Brotli middleware public interface."""
+
+    def __init__(
+        self,
+        app: ASGIApp,
+        quality: int = 4,
+        mode: str = "text",
+        lgwin: int = 22,
+        lgblock: int = 0,
+        minimum_size: int = 400,
+    ) -> None:
+        """
+        Arguments.
+        mode: The compression mode can be:
+            generic (default), text (for UTF-8 format text input)
+            or font (for WOFF 2.0).
+        quality: Controls the compression-speed vs compression-
+            density tradeoff. The higher the quality, the slower the compression.
+            Range is 0 to 11.
+        lgwin: Base 2 logarithm of the sliding window size. Range
+            is 10 to 24.
+        lgblock: Base 2 logarithm of the maximum input block size.
+            Range is 16 to 24. If set to 0, the value will be set based on the
+            quality.
+        minimum_size: Only compress responses that are bigger than this value in bytes.
+        """
+        self.app = app
+        self.quality = quality
+        self.mode = getattr(BrotliMode, mode)
+        self.minimum_size = minimum_size
+        self.lgwin = lgwin
+        self.lgblock = lgblock
```

Comment:
I don't know what to think about the options that are brotli specific... maybe we can use `kwargs` here and let the user forward what ever options they want to the Responder!

Reply:
We could use `kwargs` for most things except Mode. `BrotliResponder` needs `mode` to be an instance of `Brotli.Mode`. Or we can just make the mode always generic.

Reply:
I'm fine with generic mode, if a user wants something custom he/she can create another middleware